### PR TITLE
[OrderByComparator] LPD-28287

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPD-28287](https://liferay.atlassian.net/browse/LPD-28287)

cc: @titokenzo (resent from https://github.com/liferay-core-infra/liferay-portal/pull/7269)

[LPD-28287]: https://liferay.atlassian.net/browse/LPD-28287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by victorhcunha** [comment]
SF issue is not related to the changes and will be solved in https://github.com/brianchandotcom/liferay-portal/pull/152530

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:report:43012912

**Comment by victorhcunha** [comment]
ci:report:43068461

**Comment by victorhcunha** [comment]
ci:report:43068461

**Comment by victorhcunha** [comment]
ci:report:43068461

**Comment by Valmir-unicap** [comment]
No regression found!

File: modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/util/comparator/WorkflowDefinitionNameComparator.java [hunk 1]
```diff
@@ -20,12 +22,14 @@ public class WorkflowDefinitionNameComparator
 
 	public static final String[] ORDER_BY_FIELDS = {"name"};
 
-	public WorkflowDefinitionNameComparator() {
-		this(false);
-	}
+	public static WorkflowDefinitionNameComparator getInstance(
```
Comment (dantewang) [inline modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/util/comparator/WorkflowDefinitionNameComparator.java:25]: @victorhcunha @titokenzo The change to this file seems to be out of scope of the PR. And since it's internal with no usage, when you work on it, you don't need to deprecate it; you can just remove it.

File: modules/util/source-formatter/src/main/resources/dependencies/replacements.json [hunk 2]
```diff
@@ -124,6 +124,222 @@
...
 	},
+	{
+		"from": "new CommerceAddressCreateDateComparator(boolean paramName)",
+		"issueKey": "LPD-26935",
+		"skipParametersValidation": true,
+		"to": "CommerceAddressCreateDateComparator.getInstance(param#0#)"
+	},
+	{
+		"from": "new CommerceAddressNameComparator(boolean paramName)",
+		"issueKey": "LPD-26935",
```
Comment (dantewang) [inline modules/util/source-formatter/src/main/resources/dependencies/replacements.json:135]: @victorhcunha @titokenzo The change in this file introduces 2 ticket numbers, while the PR's ticket is LPD-28287. Is it typo?

Comment (victorhcunha) [reply modules/util/source-formatter/src/main/resources/dependencies/replacements.json]: Yes! It was a typo, I resent fixing.

File: modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/util/comparator/WorkflowDefinitionNameComparator.java (hunk 1 above)
Comment (victorhcunha) [reply modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/util/comparator/WorkflowDefinitionNameComparator.java:25]: This change is part of the PR but should be in a different commit (I resent it). By default, we're just deprecating the unused classes. After your review, I realized that some were missing this, so I added it.
But about this one, since it's not really exported, should I remove it completely? If so, I can resend it with the change.

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:stop

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:report:45693324

**Comment by dantewang** [comment]
@victorhcunha An internal unused class can be directly removed, because it's not part of the public API/SPI.

**Comment by victorhcunha** [comment]
ci:report:45693324

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
> @victorhcunha An internal unused class can be directly removed, because it's not part of the public API/SPI.

Sure! I resent the commit removing this class

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:report:46441877

**Comment by Valmir-unicap** [comment]
No regression found!

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
The SF issue is not related to the changes.

**Comment by dantewang** [comment]
ci:forward:force

**Comment by dantewang** [comment]
ci:forward:force

**Comment by victorhcunha** [comment]
ci:report:50650558

**Comment by victorhcunha** [comment]
Thanks @dantewang

**Comment by victorhcunha** [comment]
ci:forward:force
